src/selectors/layout.js

```diff
@@ -31,33 +32,46 @@ export const getGraphInput = createSelector(
 /**
  * Return the displayed width of the sidebar
  */
-export const getSidebarWidth = (visibleSidebar, outerChartWidth) => {
-  if (visibleSidebar && outerChartWidth > sidebarBreakpoint) {
-    return sidebarWidth.open;
-  }
-  return sidebarWidth.closed;
-};
+export const getSidebarWidth = visibleSidebar =>
+  visibleSidebar ? sidebarWidth.open : sidebarWidth.closed;
+
+/**
+ * Return the displayed width of the meta sidebar
+ */
+export const getMetaSidebarWidth = visibleMetaSidebar =>
+  visibleMetaSidebar ? metaSidebarWidth.open : metaSidebarWidth.closed;
 
 /**
  * Convert the DOMRect into an Object, mutate some of the properties,
  * and add some useful new ones
  */
 export const getChartSize = createSelector(
-  [getVisibleSidebar, state => state.chartSize],
-  (visibleSidebar, chartSize) => {
+  [getVisibleSidebar, getVisibleMetaSidebar, state => state.chartSize],
+  (visibleSidebar, visibleMetaSidebar, chartSize) => {
     const { left, top, width, height } = chartSize;
     if (!width || !height) {
       return {};
     }
-    const sidebarWidth = getSidebarWidth(visibleSidebar, width);
+
+    // Get the actual sidebar width
+    const sidebarWidth = getSidebarWidth(visibleSidebar);
+    const metaSidebarWidth = getMetaSidebarWidth(visibleMetaSidebar);
+
+    // Find the resulting space for the chart
+    const chartWidth = width - sidebarWidth - metaSidebarWidth;
+
+    // Chart size excludes sidebars when space is small
+    const excludeSidebars = chartWidth < chartMinWidth;
+
     return {
       left,
       top,
       outerWidth: width,
       outerHeight: height,
-      width: width - sidebarWidth,
       height,
-      sidebarWidth
+      width: excludeSidebars ? width : chartWidth,
+      sidebarWidth: excludeSidebars ? 0 : sidebarWidth,
+      metaSidebarWidth: excludeSidebars ? 0 : metaSidebarWidth
     };
```

Comment:
Yeah the idea behind that was to make the best use of available space and to keep the chart centered, given any configuration of sidebars.

Maybe that's a little too jarring in some cases then, I've changed it to always ignore both sidebars for screens `< 1200px` let me know if you think that works?

Reply:
Much better. However I'd have it still respect the main sidebar when the sidebar doesn't go full-width - otherwise at anything between 400-1200px width, the layer labels and part of the graph just sits mostly under the main sidebar:
![image](https://user-images.githubusercontent.com/1155816/97427180-24a49580-190c-11eb-8aa1-4de9b9f7c1f5.png)

also, the layer labels are never visible at <1200px widths, because they sit under the sidebar even when it's closed:
![image](https://user-images.githubusercontent.com/1155816/97428337-c37dc180-190d-11eb-84dc-83123eaafc1d.png)

Basically, I think that with respect to the main sidebar, anytime you have the graph and sidebar visible at the same time, the graph should default to the remaining screen width, because otherwise you won't see the layer labels and it looks off-centered and frankly a bit weird. Then once it hits a width where that's untenable (usually ~700px), just make the sidebar go full-width because you might as well rather than keeping it to max 400px because the graph underneath is unusable either way. This is mainly because you typically keep the main sidebar open or closed, you don't toggle it all the time.
Then for the meta panel, it should sit atop the graph content at anything below about 1200px width or so. This is because it's typically toggled often, and only temporarily open.

I might have a look into it and see if I can come up with a real code recommendation, if that's ok?

Reply:
@bru5 I've [created a new branch with some changes](https://github.com/quantumblacklabs/kedro-viz/commits/feature/pipeline-selection-styling-responsive-proposal) to show you what I'm proposing. Feel free to merge if you think this works.

I think the JS changes in particular work much better. However I've also reverted the main sidebar width breakpoint, because many datasets have nodes with long names, and on smaller screen widths (<700px) you can't see the graph properly anyway when the sidebar is open, and you don't have a mouse to hover over the node names anyway to see their titles, so why not just make it full-width anyway?
I've also removed the min-widths because they don't appear to serve any useful purpose, to me. If the screen is that small, might as well just let them shrink? Better to be useful than perfect.

Reply:
Okay thanks, I've merged your suggest changes here.

Reply:
Okay thanks! btw I hadn't yet fixed the tests that I broke by making those changes, because I wasn't sure whether you'd want to use them - if you like I can fix them, seeing as it was me who broken them?

Reply:
Just to check, with your changes it looks like the metadata panel now always overlays? So for example if you click a node on the right side of the screen, it will then be covered by the panel. Just checking that's what you intended?